toolchain/check/impl_lookup.cpp

```diff
@@ -266,6 +268,46 @@ static auto GetWitnessIdForImpl(Context& context, SemIR::LocId loc_id,
       context.sem_ir(), specific_id, impl.witness_id));
 }
 
+// In the case where `facet_const_id` is a facet, see if its facet type requires
+// that `specific_interface` is implemented. If so, return the witness from the
+// facet.
+static auto FindWitnessInFacet(Context& context, SemIR::LocId loc_id,
+                               SemIR::ConstantId facet_const_id,
+                               SemIR::SpecificInterface& specific_interface)
+    -> SemIR::InstId {
+  SemIR::InstId facet_inst_id =
+      context.constant_values().GetInstId(facet_const_id);
+  // FIXME: Should we convert from a FacetAccessType to its facet here?
+  SemIR::Inst facet_inst = context.insts().Get(facet_inst_id);
+  SemIR::TypeId facet_type_id = facet_inst.type_id();
+  if (auto facet_type_inst =
+          context.types().TryGetAs<SemIR::FacetType>(facet_type_id)) {
+    auto complete_facet_type_id = RequireCompleteFacetType(
+        context, facet_type_id, loc_id, *facet_type_inst, [&] {
+          CARBON_DIAGNOSTIC(ImplLookupForFacetWithIncompleteType, Error,
+                            "impl lookup for with incomplete facet type {0}",
+                            SemIR::TypeId);
+          return context.emitter().Build(
+              loc_id, ImplLookupForFacetWithIncompleteType, facet_type_id);
+        });
+    if (complete_facet_type_id.has_value()) {
+      const auto& complete_facet_type =
+          context.complete_facet_types().Get(complete_facet_type_id);
+      for (auto interface : complete_facet_type.required_interfaces) {
+        if (interface == specific_interface) {
+          // TODO: Need to get the right witness when there are multiple.
+          return GetOrAddInst<SemIR::FacetAccessWitness>(
+              context, loc_id,
+              {.type_id = GetSingletonType(context,
```

Comment:
```suggestion
              SemIR::FacetAccessWitness{.type_id = GetSingletonType(context,
```

I find putting the type name on the {} here instead of the function name above to be marginally more legible, wdyt?